engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -191,46 +202,63 @@ public void processAggregationAsync(Event event) {
     }
 
     public void processAggregationSync(Event event) {
-        AggregationCommand aggregationCommand;
+
+        List<AggregationCommand> aggregationCommands = new ArrayList<>();
         Timer.Sample consumedTimer = Timer.start(registry);
 
         try {
             Action action = actionParser.fromJsonString(event.getPayload());
-            Map<String, Object> map = action.getEvents().get(0).getPayload().getAdditionalProperties();
-            aggregationCommand = objectMapper.convertValue(map, AggregationCommand.class);
+            for (com.redhat.cloud.notifications.ingress.Event actionEvent : action.getEvents()) {
+                aggregationCommands.add(objectMapper.convertValue(actionEvent.getPayload().getAdditionalProperties(), AggregationCommand.class));
```

Comment:
An sub aggregation command shouldn't failed on deserialization because the message comes from Aggregator module.
If it's happends, I think must reject the entire aggregation request.
WDYT ?

Reply:
The aggregation commands won't be re-emitted in case of failures here. I would process the valid data and log the invalid one, rather than rejecting the entire data.

Reply:
Ok